convert2rhel/checks.py

```diff
@@ -318,3 +341,24 @@ def get_unsupported_kmods(host_kmods, rhel_supported_kmods):
     Ignore certain kmods mentioned in the system configs. These kernel modules moved to kernel core, meaning that the
     functionality is retained and we would be incorrectly saying that the modules are not supported in RHEL."""
     return host_kmods - rhel_supported_kmods - set(system_info.kmods_to_ignore)
+
+
+def check_rhel_compatible_kernel_is_used():
+    """Check if the kernel packages are RHEL compatible."""
+    kernel_release, _ = run_subprocess("uname -r")
+    kernel_release = kernel_release.strip()
+    if not RHEL_COMPATIBLE_KERNEL_RELEASE_RE.match(kernel_release):
+        logger.critical(
+            "The booted kernel is not compatible with the target RHEL system. "
+            "To proceed with the conversion, boot a RHEL-compatible kernel "
+            "(RHCK) by executing the following steps:\n\n"
+            "1. yum install kernel\n"
+            "2. (optional step) grubby --set-default "
+            "/boot/vmlinuz-{{ installed_kernel_version_from_step_1 }}\n"
+            "3. reboot the machine and if step 2 was not applied choose RHCK "
+            "(installed in step 1) manually."
```

Comment:
```suggestion
            "The booted kernel name does not follow the naming schema of a standard RHEL kernel. "
            "To proceed with the conversion, boot a kernel that is available in the {0} base repository"
            " by executing the following steps:\n\n"
            "1. enable the {0} base repository"
            "2. yum install kernel\n"
            "3. (optional step) grubby --set-default "
            "/boot/vmlinuz-{{ installed_kernel_version_from_step_1 }}\n"
            "4. reboot the machine and if step 2 was not applied choose the kernel"
            " installed in step 1 manually".format(system_info.name)
```

Reply:
"kernel name" does not fit to me. I mean, I've never hear calling "version & release" as kernel name. Not sure it people will understand what is meant by that. Even the naming schema would be confusing for people in this case. I would change just the first line, the rest seems ok to me. What about:
```
The booted kernel version is incompatible with the one in RHEL X or this particular kernel is not supported for the conversion.
```

Maybe in case of RT I would poitn people to support for a guidance when they need the preemptive kernel used.